kubeflow/core/kubeform_spawner.py

```diff
@@ -177,6 +177,26 @@ def _expand_user_properties(self, template):
 c.KubeSpawner.singleuser_working_dir = '/home/jovyan'
 volumes = []
 volume_mounts = []
+
+# Allow environment vars to override uid and gid.
+# This allows local host path mounts to be read/writable
+env_uid = os.environ.get('NOTEBOOK_UID')
+if env_uid and env_uid != 'null':
+    c.KubeSpawner.singleuser_uid = int(env_uid)
+env_gid = os.environ.get('NOTEBOOK_GID')
+if env_gid and env_gid != 'null':
+    c.KubeSpawner.singleuser_fs_gid = int(env_gid)
+    def modify_pod_hook(spawner, pod):
```

Comment:
Should we guard the lifecycle with its own hook? Maybe define an environment variable for the directory containing the notebooks.

I worry it might be brittle to infer this based on gid.

Reply:
Added a separate env var for this.